src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+import org.broadinstitute.hellbender.exceptions.UserException.BadArgumentValue;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+import org.reflections.ReflectionUtils;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * The class responsible for computing annotations for variants.
+ */
+public final class VariantAnnotatorEngine {
+    private final List<InfoFieldAnnotation> infoAnnotations;
+    private final List<GenotypeAnnotation> genotypeAnnotations;
+
+    private final VariantOverlapAnnotator variantOverlapAnnotator;
+
+    private VariantAnnotatorEngine(final AnnotationManager annots, final FeatureInput<VariantContext> dbSNPInput){
+        infoAnnotations = annots.createInfoFieldAnnotations();
+        genotypeAnnotations = annots.createGenotypeAnnotations();
+        variantOverlapAnnotator = initializeDBs(dbSNPInput);
+    }
+
+    /**
+     * Makes the engine for all known annotation types (minus the excluded ones).
+     * @param annotationsToExclude list of annotations to exclude
+     * @param dbSNPInput input for variants from a known set from DbSNP. The annotation engine will mark variants overlapping anything in this set using {@link htsjdk.variant.vcf.VCFConstants#DBSNP_KEY}.
+     */
+    public static VariantAnnotatorEngine ofAllMinusExcluded(final List<String> annotationsToExclude, final FeatureInput<VariantContext> dbSNPInput) {
+        Utils.nonNull(annotationsToExclude, "annotationsToExclude is null");
+        return new VariantAnnotatorEngine(AnnotationManager.ofAllMinusExcluded(annotationsToExclude), dbSNPInput);
+    }
+
+    /**
+     * Makes the engine for given annotation types and groups (minus the excluded ones).
+     * @param annotationGroupsToUse list of annotations groups to include
+     * @param annotationsToUse     list of of annotations to include
+     * @param annotationsToExclude list of annotations to exclude
+     * @param dbSNPInput input for variants from a known set from DbSNP. The annotation engine will mark variants overlapping anything in this set using {@link htsjdk.variant.vcf.VCFConstants#DBSNP_KEY}.
+     */
+    public static VariantAnnotatorEngine ofSelectedMinusExcluded(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude, final FeatureInput<VariantContext> dbSNPInput) {
+        Utils.nonNull(annotationGroupsToUse, "annotationGroupsToUse is null");
+        Utils.nonNull(annotationsToUse, "annotationsToUse is null");
+        Utils.nonNull(annotationsToExclude, "annotationsToExclude is null");
+        return new VariantAnnotatorEngine(AnnotationManager.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude), dbSNPInput);
+    }
+
+    private VariantOverlapAnnotator initializeDBs(final FeatureInput<VariantContext> dbSNPInput) {
```

Comment:
`initializeDBs()`? Perhaps this should be called `initializeOverlapAnnotator()` instead